drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -123,12 +115,11 @@ local function require_remote_pin_handler(driver, device, ib, response)
   if ib.data.value then
     --Process after all other info blocks have been dispatched to ensure MaxPINCodeLength has been processed
     device.thread:call_with_delay(0, function(t)
-      local len = device:get_latest_state("main", capabilities.lockCodes.ID, capabilities.lockCodes.maxCodeLength.NAME) or 4
-      local cred_data = math.floor(math.random() * (10 ^ len))
-      cred_data = string.format("%0" .. tostring(len) .. "d", cred_data)
-      device:set_field(lock_utils.COTA_CRED, cred_data, {persist = true})
+      generate_cota_cred_for_device(device)
+      -- delay needed to allow test to override the random credential data
       device.thread:call_with_delay(0, function(t)
-        set_cota_credential(device)
+        -- Attempt to set cota credential at the lowest index
+        set_cota_credential(device, 1)
```

Comment:
Might be a good idea to make a variable called INITIAL_COTA_INDEX or something in order to avoid the magic numbers in the code